pyiceberg/table/__init__.py

```diff
@@ -502,6 +503,71 @@ def append(self, df: pa.Table, snapshot_properties: Dict[str, str] = EMPTY_DICT)
                 for data_file in data_files:
                     append_files.append_data_file(data_file)
 
+    def _build_partition_predicate(self, spec_id: int, delete_partitions: List[Record]) -> BooleanExpression:
```

Comment:
nit: I found the `delete_partitions` argument a bit confusing here, because this function just translates a set of partition record values to its corresponding predicate. Could we rename it to something more generic to indicate that? W should also remove `spec_id` which isn't used in this function

```suggestion
    def _build_partition_predicate(self, partition_records: List[Record]) -> BooleanExpression:
```

Reply:
good catch, thanks!